temporalcli/commands.workflow_history.go

```diff
@@ -0,0 +1,104 @@
+package temporalcli
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	historypb "go.temporal.io/api/history/v1"
+	"go.temporal.io/sdk/client"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/encoding/prototext"
+)
+
+const (
+	historyFormatJSON      = "json"
+	historyFormatPB        = "pb"
+	historyFormatPBTXT     = "pbtxt"
+	historyFormatProto     = "proto"
+	historyFormatProtoText = "prototext"
+)
+
+func (c *TemporalWorkflowHistoryConvertCommand) run(cctx *CommandContext, args []string) error {
+	raw, err := os.ReadFile(c.SourceFile)
+	if err != nil {
+		return err
+	}
+
+	var history *historypb.History
+	switch c.SourceFormat.Value {
+	case historyFormatJSON:
+		history, err = client.HistoryFromJSON(bytes.NewReader(raw), client.HistoryJSONOptions{})
+		if err != nil {
+			return err
+		}
+
+	case historyFormatProtoText:
+		fallthrough
+	case historyFormatPBTXT:
+		history = new(historypb.History)
+		err = prototext.Unmarshal(raw, history)
+		if err != nil {
+			return err
+		}
+
+	case historyFormatProto:
+		fallthrough
+	case historyFormatPB:
+		history = new(historypb.History)
+		err = history.Unmarshal(raw)
+		if err != nil {
+			return err
+		}
+
+	default:
+		return fmt.Errorf("BUG: --source-format=%q is not implemented yet", c.SourceFormat.Value)
+	}
+
+	switch c.TargetFormat.Value {
+	case historyFormatJSON:
+		mo := protojson.MarshalOptions{Indent: "  "}
+		raw, err = mo.Marshal(history)
+		if err != nil {
+			return err
+		}
+
+	case historyFormatProtoText:
+		fallthrough
+	case historyFormatPBTXT:
+		mo := prototext.MarshalOptions{Indent: "  "}
+		raw, err = mo.Marshal(history)
+		if err != nil {
+			return err
+		}
+
+	case historyFormatProto:
+		fallthrough
+	case historyFormatPB:
+		raw, err = history.Marshal()
+		if err != nil {
+			return err
+		}
+
+	default:
+		return fmt.Errorf("BUG: --target-format=%q is not implemented yet", c.TargetFormat.Value)
+	}
+
+	switch c.TargetFile {
+	case "":
+		fallthrough
+	case "-":
+		_, err = cctx.Options.Stdout.Write(raw)
+		if err != nil {
+			return err
+		}
+
+	default:
+		err = os.WriteFile(c.TargetFile, raw, 0o666)
```

Comment:
```suggestion
		return os.WriteFile(c.TargetFile, raw, 0600)
```

`o` not needed (`0` prefix assumes octal), but can leave in if it reads better for you. Also, technically history can have some private data, wonder if we want owner-only.

Reply:
I prefer `0o` because I think it's what C should have used for octal all along.

As for `666` vs `600`, I'm always a little cautious about overruling the user's umask, but if you think it's reasonable here (when `temporal workflow show -o json > file.json` would just use the umask) we can do that.

Reply:
Works for me, I think what you have here is fine